ManageIQ/manageiq
# Move mime-types-redirector to fake gemdir

Automate uses a blackboxed ruby environment, and doesn't include our appliance bundle when running the code.  To make it easier for these to be added to the automate ENV and used with default rubygems, the code for the redirector has been put into a rubygems like gem dir so that it can be added to the `ENV["GEM_PATH"]` and loaded properly via rubygems.

Will require updating the `manageiq-automate_engine` repo for this to work ( https://github.com/ManageIQ/manageiq-automation_engine/pull/125 ).


Links
-----
* https://bugzilla.redhat.com/show_bug.cgi?id=1518865
* https://github.com/ManageIQ/manageiq-automation_engine/pull/125
* The PR that started it all... https://github.com/ManageIQ/manageiq/pull/14525


Steps for Testing/QA
--------------------

I was testing this doing the following:

* Create an Automate domain for testing (I called it "RestclientTesting", but doesn't really matter)
   - Create a in that domain Namespace called "System"
   - Create a class called "Request"
   - Create a method called "InspectMe"
* From `Automation -> Automate -> Simulation`, run the `InspectMe` method
   - fill in the `Request` field and press submit
* The method should work just fine without an error

Of note, you will need to probably make some changes to your environment for this to work

* Uninstall any `mime-types` gems that might still be installed in your default gem dir:
   - `$gem uninstall mime-types`
* Include the tweaks to `manageiq-automate_engine` in your `bundler.d/Gemfile.dev.rb`
   - my fork (`NickLaMuro`), and the branch should be `fix_restclient_usage_with_mime_types_redirector`
